notebooks/ir_scattered_light_manual_corrections/Correcting_for_Scattered_Light_in_IR_Exposures_by_Manually_Subtracting_Bad_Reads.ipynb

```diff
@@ -0,0 +1,966 @@
+    "dflat = dflat_im[1].data\n",
+    "\n",
+    "\n",
+    "#Computing the final error\n",
+    "\n",
+    "#### Variance terms\n",
+    "\n",
+    "## poisson error: sqrt(signal), flux = final_sci, dark is converted to electrons\n",
+    "signal = (final_sci*(pflat*dflat)+final_dark*gain_2D)\n",
+    "## flat errors\n",
+    "dflat_err=dflat_im['ERR'].data\n",
+    "pflat_err=pflat_im['ERR'].data\n",
+    "#dark error\n",
+    "dark_err = (dark_im['ERR'].data)*gain_2D \n",
+    "##final\n",
+    "final_err = np.sqrt(RN**2 + signal + (dark_err**2) + (pflat_err*final_sci*dflat)**2\\\n",
+    "                  + (dflat_err*final_sci*pflat)**2)\n",
+    "    \n",
+    "final_err /= (pflat*dflat*final_time)\n",
+    "final_err[np.isnan(final_err)] = 0\n",
+    "\n",
+    "#and finally the final science image (converted back to count rate)\n",
+    "final_sci /= final_time      "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### STEP 4: Update the FLT Product"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we update the FLT product created in step 1 with the new science array, error array, and header keywords (such as the new total exposure time)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "flt = fits.open(ima_filepath_new.replace('ima', 'flt'), mode='update')\n",
+    " \n",
+    "#Updating the flt data\n",
+    "\n",
+    "flt['SCI'].data = final_sci[5:-5,5:-5]\n",
+    "flt['ERR'].data = final_err[5:-5,5:-5]\n",
+    "\n",
+    "\n",
+    "# Updating the FLT header\n",
+    "flt[0].header['IMA2FLT'] = (1, 'FLT extracted from IMA file')\n",
+    "flt[0].header['EXPTIME'] = np.max(final_time)\n",
+    "flt[0].header['NPOP'] = (len(bad_reads), 'Number of reads popped from the sequence')\n",
+    "\n",
+    "flt.flush()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Putting It All Together"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below, we have included the entire correction process (steps 1-4 above) in a function, for convenience. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def remove_reads(raw_filepath, bad_reads = []):\n",
+    "    '''\n",
+    "    From the final IMA read, subtract data of reads affected by anomalies.\n",
+    "    Compute a corrected science image and error image, and place them in the FLT product. \n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    raw_filepath: str\n",
+    "        Path to a RAW full-frame IR image fits file.\n",
+    "        \n",
+    "    bad_reads: list of int\n",
+    "        List of the IMSET numbers (science extension numbers) of reads affected by anomalies.\n",
+    "    '''\n",
```

Comment:
Define `ima_filepath` and `flt_filepath` to use throughout the function (the former was not defined in the function so the function would've crashed towards the end where you call it):
```
flt_filepath = raw_filepath.replace('raw', 'flt')
ima_filepath = raw_filepath.replace('raw', 'ima')
```